tests/importer/importer.cpp

```diff
@@ -0,0 +1,761 @@
+    auto localConcreteUnits = libcellml::Model::create("localConcreteUnits");
+    localConcreteUnits->addUnits(libcellml::Units::create("units5"));
+
+    // Add all the concrete models to the Importer. These don't need resolving as they have no imports of their own.
+    auto importer = libcellml::Importer::create();
+    importer->addModel(concreteUnits, "i_dont_exist_yet.cellml"); // add with key that's not a file (yet).
+    importer->addModel(concreteComponents, "https://www.example.com/myComponents.cellml"); // add with a key that is a remote URL
+
+    // Add models which have imports to the Importer by resolving them. They are automatically saved into the
+    // Importer's library, along with any dependencies, under the names used to resolve them.
+    importer->resolveImports(importedUnits, resourcePath("importer/"));
+    importer->resolveImports(importedComponents, resourcePath("importer/"));
+
+    EXPECT_FALSE(importedUnits->hasUnresolvedImports());
+    EXPECT_FALSE(importedComponents->hasUnresolvedImports());
+
+    // Now create a funky model that imports from all over the place and see what happens.
+    std::string funkyString = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                              "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"funky\">\n"
+                              // Status quo functionality, key is local URL + base file path as the local URL doesn't exist as a key.
+                              "  <import xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"units_imported.cellml\">\n"
+                              "    <units units_ref=\"units1_imported\" name=\"units1FromLibrary\"/>\n"
+                              "  </import>\n"
+                              // Use the local URL for a hidden child source component.
+                              "  <import xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"components_source.cellml\">\n"
+                              "    <component component_ref=\"component1\" name=\"component1FromLibrary\"/>\n"
+                              "  </import>\n"
+                              // Use the absolute URL for a hidden child source component.
+                              "  <import xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\""
+                              + resourcePath("importer/components_source.cellml")
+                              + "\">\n"
+                                "    <component component_ref=\"component2\" name=\"component2FromLibrary\"/>\n"
+                                "  </import>\n"
+                                // Remote website URL used for key.
+                                "  <import xmlns:xlink=\"http://www.w3.org/1999/xlink\" xlink:href=\"https://www.example.com/myComponents.cellml\">\n"
+                                "    <component component_ref=\"component3\" name=\"component3FromLibrary\"/>\n"
+                                "  </import>\n"
```

Comment:
How do we expect the importer to resolve this?

Reply:
It's there for future-proofing: all the `Importer` needs is a unique string, so when we're able to read remote files, the `Importer` construction doesn't need to change.

Reply:
I take this one back, I can see that there is an entry for `https://www.example.com/myComponents.cellml` in the importer.